### ISSUE TITLE
aioredis integration: Tracing breaks pipeline as context managers

### ISSUE THREAD

**Issue by mklokocka** [DataDog/dd-trace-py#3106]
Hello,

looks like the `aioredis` integration breaks the interface of `Pipeline` objects as context managers:

```py
RuntimeWarning: coroutine 'traced_pipeline' was never awaited
```

and

```py
async with redis.pipeline(transaction=True) as pipe:
AttributeError: __aexit__
```

This is bad since the documented of usage is exactly as context managers (see https://aioredis.readthedocs.io/en/latest/migration/#pipelines-and-transactions-multiexec).

The fix for now is to just use pipelines outside of contexts, without relying on them as context managers, but that is less than ideal.

`ddtrace` is the latest version (`0.57.0`).

**Comment by brettlangdon** [comment]
@mklokocka thank you for reporting. I have been able to reproduce.

```python
# run.py
import asyncio

import aioredis


async def main():
    redis = await aioredis.from_url("redis://localhost")
    async with redis.pipeline(transaction=True) as pipe:
        ok1, ok2 = await (pipe.set("key1", "value1").set("key2", "value2").execute())
    assert ok1
    assert ok2


if __name__ == "__main__":
    asyncio.run(main())
```

```
$ ddtrace-run python run.py
```

**Comment by brettlangdon** [comment]
I have opened up #3108 with the fix for this issue.

Once it is reviewed and merged I will backport and release in a `0.57.1` release.

**Comment by brettlangdon** [comment]
This has been released in `ddtrace==0.57.1`
